### ISSUE TITLE
Clearing the logfile via the GUI and via a hotkey

### ISSUE THREAD

**Issue by DrSooom** [nvaccess/nvda#9279]
### Original suggestion by @AbhitejJohn:
Quote from [his comment on issue #9275](https://github.com/nvaccess/nvda/issues/9275#issuecomment-463547835):
> Also I wish I could clear/reset the logs from the UI :)

### Path:
NVDA menu » Tools » Log Viewer...
### Steps to reproduce:
1. Change the log level to "debug" in the General NVDA Settings.
2. Open the Log Viewer via the NVDA menu.
3. Try to remember the actual last line.
4. Open the start menu. Or just do anything else. It doesn't matter. But its logs are very, very important for you.
5. Switch back to the Log Viewer.
6. Have fun by finding the previous remembered line.
### Actual behaviour:
It isn't possible to clear the logged content until step 4. It has to deleted with Notepad++ or with any other useful source code editor manually, after the logfile was saved first.
### Expected behaviour:
+ Add a new menu item called "Clear" below "Save as..." in the log menu in the Log Viewer.
  + Activating it will delete the whole content of the current logfile.
  + If this isn't possible, than the file "nvda.log" itself should be deleted and "recreated" directly after that.
  + And in both case, the Log Viewer content has to be updated (F5) automatically.
+ Allocate CTRL+D for this new item.
+ btw: Allocate CTRL+Q for the item "Exit" as well. (I know that the Log Viewer can be closed with ESC, but there is no hotkey mentioned for this menu item (in the German GUI).)
+ Allocate NVDA+Shift+F1 to clear the logfile without having to open the log Viewer.
#### NVDA installed/portable:
Both
#### NVDA version:
2018.4.1
#### Windows version:
Win7 to Win10-1903

**Comment by Brian1Gaff** [comment]
Would not this need to close the log, then suspend writing to it, rename the 
old log and  start a new one and then write what you buffered to that new 
log?
 Its a bad enough nightmare when nvda has to restart after a lot of copying 
and renaming of files that it can clog up with a windows error as a file is 
still  locked. That was why what I did was to use an external batch file to 
do it all.
 Brian

@echo off
cd "%temp%"
start /WAIT nvda.exe -q

start notepad.exe nvda.log
rem, this line just kicks its heels for a bit to make sure the log is loaded 
into notepad before rebooting nvda.
ping -n 5 127.0.0.1>nul
start nvda.exe

exit

===
Note that this also allows the old log to be created.


Of course it does not preserve the start and end bits, but is that really a 
major issue if you know what to hunt for?

Brian



bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
-----

**Comment by Adriani90** [comment]
This could just be a menu item in the protocol menu of the log file.
